src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -180,6 +180,14 @@ private function mapRequestPayload(Request $request, string $type, MapRequestPay
             throw new UnsupportedMediaTypeHttpException(sprintf('Unsupported format, expects "%s", but "%s" given.', implode('", "', (array) $attribute->acceptFormat), $format));
         }
 
+        if ('array' === $type) {
+            if (null === $attribute->itemsType) {
+                throw new \LogicException(sprintf('For proper deserialization, the %s::$itemsType value is required when the argument type is "array".', MapRequestPayload::class));
```

Comment:
should we throw when itemsType is set but the type is not array?
also, what about renaming itemsType to just "type"?

Reply:
>should we throw when itemsType is set but the type is not array?

Added the 2nd exception and the checks were moved earlier to the `resolve()` moment.

>also, what about renaming itemsType to just "type"?

It works for me. Renamed.

Reply:
I also added more tests to cover these exceptions